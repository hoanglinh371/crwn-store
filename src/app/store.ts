import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';

import cartReducer from '../features/cart/redux/cart.slice';
import categoryReducer from '../features/category/redux/category.slice';
import productReducer from '../features/product/redux/product.slice';

export const store = configureStore({
  reducer: {
    productReducer,
    cartReducer,
    categoryReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
