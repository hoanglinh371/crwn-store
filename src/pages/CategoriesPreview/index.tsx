import React from 'react';
import CategoryPreview from 'src/components/CategoryPreview';

import SHOP_DATA from 'src/shop-data';

const CategoriesPreview: React.FC = () => {
  return (
    <React.Fragment>
      {Object.keys(SHOP_DATA).map((title) => {
        const products = SHOP_DATA[title];
        return (
          <CategoryPreview key={title} title={title} products={products} />
        );
      })}
    </React.Fragment>
  );
};

export default CategoriesPreview;
