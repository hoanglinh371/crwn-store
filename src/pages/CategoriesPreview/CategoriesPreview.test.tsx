import { cleanup, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import { store } from 'src/app/store';
import CategoriesPreview from '.';

describe('CategoriesPreview component', () => {
  afterEach(cleanup);

  it('Should_MatchSnapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <CategoriesPreview />
        </BrowserRouter>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
