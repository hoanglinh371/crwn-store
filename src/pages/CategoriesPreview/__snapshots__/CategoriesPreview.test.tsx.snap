// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesPreview component Should_MatchSnapshot 1`] = `
<div>
  <div
    class="container"
  >
    <h2
      class="title"
    >
      <a
        href="/shop/hats"
      >
        HATS
      </a>
    </h2>
    <div
      class="preview"
    >
      <div
        class="container"
      >
        <img
          alt="Brown Brim"
          src="https://i.ibb.co/ZYW3VTp/brown-brim.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Brown Brim
          </span>
          <span
            class="price"
          >
            $
            25
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
      <div
        class="container"
      >
        <img
          alt="Blue Beanie"
          src="https://i.ibb.co/ypkgK0X/blue-beanie.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Blue Beanie
          </span>
          <span
            class="price"
          >
            $
            18
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
      <div
        class="container"
      >
        <img
          alt="Brown Cowboy"
          src="https://i.ibb.co/QdJwgmp/brown-cowboy.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Brown Cowboy
          </span>
          <span
            class="price"
          >
            $
            35
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
      <div
        class="container"
      >
        <img
          alt="Grey Brim"
          src="https://i.ibb.co/RjBLWxB/grey-brim.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Grey Brim
          </span>
          <span
            class="price"
          >
            $
            25
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <h2
      class="title"
    >
      <a
        href="/shop/sneakers"
      >
        SNEAKERS
      </a>
    </h2>
    <div
      class="preview"
    >
      <div
        class="container"
      >
        <img
          alt="Adidas NMD"
          src="https://i.ibb.co/0s3pdnc/adidas-nmd.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Adidas NMD
          </span>
          <span
            class="price"
          >
            $
            220
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
      <div
        class="container"
      >
        <img
          alt="Adidas Yeezy"
          src="https://i.ibb.co/dJbG1cT/yeezy.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Adidas Yeezy
          </span>
          <span
            class="price"
          >
            $
            280
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
      <div
        class="container"
      >
        <img
          alt="Black Converse"
          src="https://i.ibb.co/bPmVXyP/black-converse.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Black Converse
          </span>
          <span
            class="price"
          >
            $
            110
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
      <div
        class="container"
      >
        <img
          alt="Nike White AirForce"
          src="https://i.ibb.co/1RcFPk0/white-nike-high-tops.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Nike White AirForce
          </span>
          <span
            class="price"
          >
            $
            160
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <h2
      class="title"
    >
      <a
        href="/shop/jackets"
      >
        JACKETS
      </a>
    </h2>
    <div
      class="preview"
    >
      <div
        class="container"
      >
        <img
          alt="Black Jean Shearling"
          src="https://i.ibb.co/XzcwL5s/black-shearling.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Black Jean Shearling
          </span>
          <span
            class="price"
          >
            $
            125
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
      <div
        class="container"
      >
        <img
          alt="Blue Jean Jacket"
          src="https://i.ibb.co/mJS6vz0/blue-jean-jacket.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Blue Jean Jacket
          </span>
          <span
            class="price"
          >
            $
            90
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
      <div
        class="container"
      >
        <img
          alt="Grey Jean Jacket"
          src="https://i.ibb.co/N71k1ML/grey-jean-jacket.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Grey Jean Jacket
          </span>
          <span
            class="price"
          >
            $
            90
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
      <div
        class="container"
      >
        <img
          alt="Brown Shearling"
          src="https://i.ibb.co/s96FpdP/brown-shearling.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Brown Shearling
          </span>
          <span
            class="price"
          >
            $
            165
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <h2
      class="title"
    >
      <a
        href="/shop/womens"
      >
        WOMENS
      </a>
    </h2>
    <div
      class="preview"
    >
      <div
        class="container"
      >
        <img
          alt="Blue Tanktop"
          src="https://i.ibb.co/7CQVJNm/blue-tank.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Blue Tanktop
          </span>
          <span
            class="price"
          >
            $
            25
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
      <div
        class="container"
      >
        <img
          alt="Floral Blouse"
          src="https://i.ibb.co/4W2DGKm/floral-blouse.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Floral Blouse
          </span>
          <span
            class="price"
          >
            $
            20
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
      <div
        class="container"
      >
        <img
          alt="Floral Dress"
          src="https://i.ibb.co/KV18Ysr/floral-skirt.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Floral Dress
          </span>
          <span
            class="price"
          >
            $
            80
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
      <div
        class="container"
      >
        <img
          alt="Red Dots Dress"
          src="https://i.ibb.co/N3BN1bh/red-polka-dot-dress.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Red Dots Dress
          </span>
          <span
            class="price"
          >
            $
            80
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <h2
      class="title"
    >
      <a
        href="/shop/mens"
      >
        MENS
      </a>
    </h2>
    <div
      class="preview"
    >
      <div
        class="container"
      >
        <img
          alt="Camo Down Vest"
          src="https://i.ibb.co/xJS0T3Y/camo-vest.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Camo Down Vest
          </span>
          <span
            class="price"
          >
            $
            325
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
      <div
        class="container"
      >
        <img
          alt="Floral T-shirt"
          src="https://i.ibb.co/qMQ75QZ/floral-shirt.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Floral T-shirt
          </span>
          <span
            class="price"
          >
            $
            20
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
      <div
        class="container"
      >
        <img
          alt="Black & White Longsleeve"
          src="https://i.ibb.co/55z32tw/long-sleeve.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Black & White Longsleeve
          </span>
          <span
            class="price"
          >
            $
            25
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
      <div
        class="container"
      >
        <img
          alt="Pink T-shirt"
          src="https://i.ibb.co/RvwnBL8/pink-shirt.png"
        />
        <div
          class="footer"
        >
          <span
            aria-label="name"
            class="name"
          >
            Pink T-shirt
          </span>
          <span
            class="price"
          >
            $
            25
          </span>
        </div>
        <button
          aria-label="button"
          class="container inverted"
          type="button"
        >
          Add To Cart
        </button>
      </div>
    </div>
  </div>
</div>
`;
